From patchwork Thu Jan 17 11:45:00 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Dmitry Monakhov <dmonakhov@openvz.org>
X-Patchwork-Id: 10767959
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 922446C2
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 17 Jan 2019 11:53:24 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 813632E73B
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 17 Jan 2019 11:53:24 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 757222EB97; Thu, 17 Jan 2019 11:53:24 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 5183E2EB93
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 17 Jan 2019 11:53:23 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726002AbfAQLxV (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 17 Jan 2019 06:53:21 -0500
Received: from forwardcorp1j.cmail.yandex.net ([5.255.227.105]:40080 "EHLO
        forwardcorp1j.cmail.yandex.net" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S1725872AbfAQLxV (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 17 Jan 2019 06:53:21 -0500
X-Greylist: delayed 496 seconds by postgrey-1.27 at vger.kernel.org;
 Thu, 17 Jan 2019 06:53:20 EST
Received: from mxbackcorp1g.mail.yandex.net (mxbackcorp1g.mail.yandex.net
 [IPv6:2a02:6b8:0:1402::301])
        by forwardcorp1j.cmail.yandex.net (Yandex) with ESMTP id 01A4420B3C
        for <kvm@vger.kernel.org>; Thu, 17 Jan 2019 14:45:03 +0300 (MSK)
Received: from smtpcorp1p.mail.yandex.net (smtpcorp1p.mail.yandex.net
 [2a02:6b8:0:1472:2741:0:8b6:10])
        by mxbackcorp1g.mail.yandex.net (nwsmtp/Yandex) with ESMTP id
 MgiURjWqw9-j2RiKk2h;
        Thu, 17 Jan 2019 14:45:02 +0300
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=yandex-team.ru;
 s=default;
        t=1547725502; bh=iLM0E2sxvoHDJDl5v5cAvc/aGGX9Qgdo3xXAbPeDX1A=;
        h=From:To:Cc:Subject:Date:Message-Id;
        b=ItgG0ibxzEnC2tVJ80fIJ1IJPBRodz2P+swxod3D+1NMU2wKq9vEaUeVBNKo+TD1S
         6IXISzZqPv0mYv2YuBanlTM6LRbkAy8gsC7Km9iorJApOWzsfRs5mAPhJQD9kXsIzp
         WNAXin82tntv6eds38GzM89OgrYpAxe/nP+LcYCk=
Authentication-Results: mxbackcorp1g.mail.yandex.net;
 dkim=pass header.i=@yandex-team.ru
Received: from 95.108.173.73-red.dhcp.yndx.net
 (95.108.173.73-red.dhcp.yndx.net [95.108.173.73])
        by smtpcorp1p.mail.yandex.net (nwsmtp/Yandex) with ESMTPSA id
 S0WjyZbSwi-j2VeWRvJ;
        Thu, 17 Jan 2019 14:45:02 +0300
        (using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
        (Client certificate not present)
From: Dmitry Monakhov <dmonakhov@openvz.org>
To: kvm@vger.kernel.org
Cc: Dmitry Monakhov <dmtrmonakhov@yandex-team.ru>
Subject: [PATCH kvmtool] init: fix sysfs mount arguments
Date: Thu, 17 Jan 2019 11:45:00 +0000
Message-Id: <20190117114500.9702-1-dmonakhov@openvz.org>
X-Mailer: git-send-email 2.18.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Dmitry Monakhov <dmtrmonakhov@yandex-team.ru>

It is not good idea to pass empty 'source' argument to  mount(2) because libmount complains about
incorrect /proc/self/mountinfo structure. This affect many applications such as findmnt, umount and etc.
Let's add fake source argument to sysfs mount command as we do with all other filesystems.

#TESTCASE_BEGIN:
## findmount will complain like this 'findmnt: /proc/self/mountinfo: parse error at line 4 -- ignored'
echo "/bin/findmnt" > ./test.sh
./lkvm setup init-mount-test
./lkvm run -d init-mount-test --sandbox test.sh
rm -rf ~/.lkvm/init-mount-test
#TESTCASE_END
---
 guest/init.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/guest/init.c b/guest/init.c
index 46e3fa4..22fe765 100644
--- a/guest/init.c
+++ b/guest/init.c
@@ -32,8 +32,8 @@ static void do_mounts(void)
 {
 #ifndef CONFIG_GUEST_PRE_INIT
 	mount("hostfs", "/host", "9p", MS_RDONLY, "trans=virtio,version=9p2000.L");
-#endif
-	mount("", "/sys", "sysfs", 0, NULL);
+#endif	
+	mount("sysfs", "/sys", "sysfs", 0, NULL);
 	mount("proc", "/proc", "proc", 0, NULL);
 	mount("devtmpfs", "/dev", "devtmpfs", 0, NULL);
 	mkdir("/dev/pts", 0755);
